syntax = "proto3";
import "zss_geometry.proto";
package ZSS.New;

message Debug_Robot{
	Point pos = 1;
	float dir = 2;
}
message Debug_Line{
	Point start = 1;
	Point end = 2;
	bool BACK = 3;
	bool FORWARD = 4;
}
message Debug_Arc{
	Rectangle rect = 1;
	float start = 2;
	float span = 3;
}
message Debug_Polygon{
	repeated Point vertex = 1;
}
message Debug_Text{
	Point pos = 1;
	string text = 2;
	float size = 3;
	int32 weight = 4;
}
message Debug_CubicBezier{
	Point start = 1;
	Point c1 = 2;
	Point c2 = 3;
	Point end = 4;
}
message Debug_Points{
	repeated Point point = 1;
}

message Debug_Msg {
	enum Debug_Type {
		ROBOT 	= 0;
		LINE 	= 1;		
		ARC 	= 2;
		POLYGON = 3;
		TEXT 	= 4;
		BEZIER  = 5;
		POINTS  = 6;
	}
	enum Color {
		WHITE	= 0;
		RED		= 1;
		ORANGE	= 2;
		YELLOW	= 3;
		GREEN	= 4;
		CYAN	= 5;
		BLUE	= 6;
		PURPLE	= 7;
		GRAY	= 8;
		BLACK	= 9;
		USE_RGB	= 10;
	}
	Debug_Type type = 1;
	Color color = 2;
	Debug_Robot robot = 3;
	Debug_Line line = 4;
	Debug_Arc arc = 5;
	Debug_Polygon polygon = 6;
	Debug_Text text = 7;
	Debug_CubicBezier bezier = 8;
	Debug_Points points = 9;
	int32 RGB_value = 10;
}
message Debug_Msgs{
	repeated Debug_Msg msgs = 1;
}
message Debug_Heatmap{
	message Heat{
		int32 value = 1;
		repeated Point p = 2;
	}
	repeated Heat points = 1;
}
